analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/Assignment.cs

```diff
@@ -22,15 +22,15 @@
 
 namespace SonarAnalyzer.SymbolicExecution.Roslyn.OperationProcessors
 {
-    internal static class SimpleAssignment
+    internal static class Assignment
     {
         public static ProgramState Process(SymbolicContext context, ISimpleAssignmentOperationWrapper assignment)
         {
             var rightSide = context.State[assignment.Value];
             var newState = context.State
                 .SetOperationValue(assignment.Target, rightSide)
                 .SetOperationValue(assignment.WrappedOperation, rightSide);
-            return assignment.Target.TrackedSymbol() is { } symbol
+            return newState.ResolveCapture(assignment.Target).TrackedSymbol() is { } symbol
```

Comment:
This seems unrelated. Is it covered by a test?
Educational: "Capture" refers to flow capture, doesn't it? It is unrelated to variables captured by lambdas and alike, right? If so, a flow capture is referring to temporary variables introduced by lowering the code, doesn't it:
```cs
a?.ToString()?.ToString();
// lowered
if (a != null)
{
    string text = a.ToString(); // text is a flowcapture?
    if (text != null)
    {
        text.ToString();
    }
}

Reply:
It is related, `??=` doesn't work without it.
Yes, capture is related to flow capture. Effectively a temporary pointer to a symbol. 
In this case, actually
`a` will be one capture
`a.ToString()` another
`text` yet another
and `text != null` yet another if I'm not mistaken.

What's missing in your lowering schema is the last step, that is
```
final capture = result of the final operation
```
This is an assignment operation. And it assigns to a capture of `a` and that's exactly why we need this `ResolveCapture` here.